lib_loader.go

```diff
@@ -0,0 +1,81 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Purego only works on linux/macOS with amd64 and arm64 from now
+//go:build (linux || darwin) && (amd64 || arm64) && !cgo
+
+package waf
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/ebitengine/purego"
+)
+
+// LibLoader is created to wraps all interactions with the purego library
+type LibLoader struct {
```

Comment:
Sorry, I'm realising only now that this "loader" naming is not correct: this is not a loader, but rather its result.
I suggest you rename `{lib,waf}Loader` into `{lib,waf}Handle` or similar 🙏

Reply:
This will be very confusing with the waf Handle returned by ddwaf_init

Reply:
And I am not ok with the 'handle' name to begin with. What do you think of `libDl` and `wafDl` (dl as in dynamic library) ?